server/src/main/java/org/elasticsearch/index/mapper/DocumentParser.java

```diff
@@ -674,7 +681,9 @@ private static void parseDynamicValue(final DocumentParserContext context, Strin
             failIfMatchesRoutingPath(context, currentFieldName);
             return;
         }
-        context.dynamic().getDynamicFieldsBuilder().createDynamicFieldFromValue(context, currentFieldName);
+        if (context.dynamic().getDynamicFieldsBuilder().createDynamicFieldFromValue(context, currentFieldName) == false) {
+            failIfMatchesRoutingPath(context, currentFieldName);
```

Comment:
can you expand on this additional failIfMatchesRoutingPath call? and the one above as well?